File: python/src/wdl_model/model/types/wdl_type_reference_type.py
from __future__ import annotations

from .wdl_type import ComponentType, WdlType


class WdlTypeReferenceType(WdlType):
    def __init__(self, referencedTypeName: str | None = None, optional: bool = False):
        super().__init__(optional)
        self.referencedTypeName = referencedTypeName

    def componentType(self) -> ComponentType:
        return ComponentType.TYPE_REFERENCE

    def referenceName(self) -> str | None:
        return self.referencedTypeName

    def setReferenceName(self, referencedTypeName: str) -> None:
        self.referencedTypeName = referencedTypeName

File: typescript/src/model/types/wdl-type-reference-type.ts
/** User-defined type reference nodes. */
import { WdlType, WdlTypeComponentType } from './wdl-type.js';

/** Models a reference to a named user-defined type such as a struct or enum. */
export class WdlTypeReferenceType extends WdlType {
  /** Creates a type reference from its target type name and optional flag. */
  public constructor(
    private referencedTypeName = '',
    optional = false,
  ) {
    super(optional);
  }

  /** Returns the broad type family for this node. */
  public componentType(): WdlTypeComponentType {
    return WdlTypeComponentType.TYPE_REFERENCE;
  }

  /** Returns the referenced type name. */
  public referenceName(): string {
    return this.referencedTypeName;
  }

  /** Sets the referenced type name. */
  public setReferenceName(referencedTypeName: string): void {
    this.referencedTypeName = referencedTypeName;
  }
}
